app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -152,6 +153,21 @@ public void onFilterApplied(Filter filter) {
         applyFilters();
     }
 
+    /**
+     * Resets filters of a specific class by removing all instances of that class from the applied
+     * filters list and then re-applies the remaining filters.
+     *
+     * @param filterClass The class of filter to reset.
+     */
+    public void onFilterReset(Class<?> filterClass) {
+        for (Filter filter : appliedFilters) {
```

Comment:
I believe we could just call something along the lines of appliedFilters.remove(filterinstance) no? Because appliedFilters is a Set not a List. See corresponding comment for what we would pass at the call location.

Reply:
Also, below line 99 I believe we should have applyFilters(); being called

Reply:
See my response about passing `this` to the callback.
It could be possible if we store the `MakeFilter` in the `MakeFilterFragment` as an attribute and then pass that `MakeFilter` to the callback